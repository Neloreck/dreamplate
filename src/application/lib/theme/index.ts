export * from "@/lib/theme/types";
export * from "@/lib/theme/utils";
