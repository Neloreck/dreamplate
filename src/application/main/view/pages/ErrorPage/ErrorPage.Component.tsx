import { Bind, Consume } from "dreamstate";
import { PureComponent, ReactNode } from "react";
import { WithSheet } from "react-jss";

// Lib.
import { Styled, Wrapped } from "@Lib/decorators";

// Data.
import { IRouterContext, routerContextManager } from "@Main/data/store";

// View.
import { IMainHeaderInjectedProps, MainHeader } from "@Main/view/components/MainHeader";
import { GlobalProvider } from "@Main/view/layouts";
import { errorPageStyle } from "./ErrorPage.Style";

// Types.
export interface IErrorPageOwnProps {}

export interface IErrorPageInjectedProps extends WithSheet<typeof errorPageStyle>, IRouterContext {}

export interface IErrorPageProps extends IErrorPageOwnProps, IErrorPageInjectedProps {}

@Wrapped(GlobalProvider)
@Styled(errorPageStyle)
@Consume(routerContextManager)
export class ErrorPage extends PureComponent<IErrorPageProps> {

  public render(): ReactNode {

    const { classes } = this.props;

    return (
      <>

        <MainHeader {...{} as IMainHeaderInjectedProps}/>

        <main className={classes.content}>

          <div className={classes.label}>
            Page not found.
          </div>

          <button onClick={this.onHomeNavigated}>
            Home
          </button>

        </main>

      </>
    );
  }

  @Bind()
  private onHomeNavigated(): void {

    const { routingActions } = this.props;

    routingActions.push("/home");
  }

}
