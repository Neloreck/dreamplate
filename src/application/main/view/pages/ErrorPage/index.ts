export { ErrorPage, IErrorPageProps } from "./ErrorPage.component";
