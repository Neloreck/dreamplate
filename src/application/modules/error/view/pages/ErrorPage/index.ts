/**
 * @packageDocumentation
 * @module @application/error
 */

export * from "./ErrorPage.component";
