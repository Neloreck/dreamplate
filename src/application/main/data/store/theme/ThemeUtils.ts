import { IApplicationTheme } from "@Main/data/store/theme/ThemeTypes";

export const createTheme = ({ palette, spacing }: IApplicationTheme): IApplicationTheme => {

  const isDark: boolean = (palette.type === "dark");

  return {
    palette: {
      ...palette,
      background: {
        default: isDark ? "#333433" : "#ffffff",
        paper: isDark ? "#5a5a5a" : "#f3f3f3"
      },
      text: {
        primary: isDark ? "#fff" : "#000",
        secondary: isDark ? "#AAA" : "#333"
      }
    },
    spacing: {
      ...spacing
    }
  };
};
