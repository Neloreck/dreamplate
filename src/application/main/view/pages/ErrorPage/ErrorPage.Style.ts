import { IApplicationTheme } from "@Main/data/store/theme/ThemeTypes";

export const errorPageStyle = ({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    flexWrap: "nowrap",
    justifyContent: "center",
    overflowY: "auto",
    width: "100%"
  },
  root: {
    backgroundColor: palette.background.paper,
    display: "flex",
    flexDirection: "column",
    height: "100%",
    width: "100%",
    transitionDuration: "250ms"
  }
});
