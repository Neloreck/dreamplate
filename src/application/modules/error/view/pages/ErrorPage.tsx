import { useManager } from "dreamstate";
import { ReactElement } from "react";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";

import { RouterManager } from "@/core/data/store";
import { MainHeader } from "@/core/view/components";
import { IApplicationTheme } from "@/lib/theme";

export const useStyles = createUseStyles(({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    alignSelf: "normal",
    backgroundColor: palette.background.default,
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    flexWrap: "nowrap",
    justifyContent: "center",
    overflowY: "auto",
    transitionDuration: "500ms",
  },
  labelCard: {
    backgroundColor: palette.secondary.light,
  },
}));

export function ErrorPage({
  classes: { content, labelCard } = useStyles(),
  routerContext: { history } = useManager(RouterManager),
}): ReactElement {
  return (
    <>
      <MainHeader />

      <main className={content}>
        <div className={labelCard}>
          {IS_DEV ? "Page was not found or current path is not serving by dev server." : "Page was not found."}
        </div>

        <Link to={history.createHref("/home")} reloadDocument>
          <button> Go Home </button>
        </Link>
      </main>
    </>
  );
}
