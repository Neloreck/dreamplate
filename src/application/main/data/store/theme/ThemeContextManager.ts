import { Bind, ContextManager } from "dreamstate";

// Lib.
import { Logger } from "@Lib/utils";

// Data.
import { EThemeType, IApplicationTheme } from "@Main/data/store/theme/ThemeTypes";
import { createTheme } from "@Main/data/store/theme/ThemeUtils";

export interface IThemeContext {
  themeActions: {
    isDark(): boolean;
    toggleTheme(): void;
  };
  themeState: {
    theme: IApplicationTheme;
  };
}

export class ThemeContextManager extends ContextManager<IThemeContext> {

  private static readonly DEFAULT_THEME_PARAMS: IApplicationTheme = {
    palette: {
      background: {
        default: "#FFF",
        paper: "#EEE"
      },
      primary: {
        dark: "#4c497c",
        light: "#9690FC",
        main: "#a8a5ec",
      },
      secondary: {
        dark: "#6b5b71",
        light: "rgb(166,143,176)",
        main: "#cfb3dc",
      },
      text: {
        primary: "#000",
        secondary: "#333"
      },
      type: EThemeType.LIGHT
    },
    spacing: {
      unit: 8
    }
  };

  protected context: IThemeContext = {
    themeActions: {
      isDark: this.isDark,
      toggleTheme: this.toggleTheme
    },
    themeState: {
      theme: createTheme(ThemeContextManager.DEFAULT_THEME_PARAMS)
    }
  };

  private readonly log: Logger = new Logger("[üç¨THEME]");
  private readonly setState = ContextManager.getSetter(this, "themeState");

  protected onProvisionStarted(): void {
    this.log.info("Started theme context provision.");
  }

  @Bind()
  private isDark(): boolean {
    return this.context.themeState.theme.palette.type === "dark";
  }

  @Bind()
  private toggleTheme(): void {

    const { theme } = this.context.themeState;

    const nextThemeType: EThemeType = (theme.palette.type === EThemeType.LIGHT ? EThemeType.DARK : EThemeType.LIGHT);

    theme.palette.type = nextThemeType;

    this.log.info(`Toggle to '${nextThemeType}'.`);

    this.setState({ theme: createTheme(theme) });
  }

}
