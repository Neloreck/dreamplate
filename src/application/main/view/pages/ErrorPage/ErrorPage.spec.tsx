// Test.
import { getThemedComponent, withMockedTheme } from "@Lib/theme/testing";
import { shallow } from "enzyme";

// View.
import { ErrorPage, IErrorPageInjectedProps } from "./ErrorPage.component";

describe("Error page component rendering.", () => {

  it("Should match HOC snapshot.", () => {

    const wrapper = shallow(<ErrorPage {...{} as IErrorPageInjectedProps}/>);

    expect(wrapper).toMatchSnapshot("Default HOC layout.");
  });

  it("Should match mounted snapshot with default theme.", () => {

    const wrapper = shallow(withMockedTheme(<ErrorPage {...{} as IErrorPageInjectedProps}/>));
    const mounted = getThemedComponent(wrapper);

    expect(mounted).toMatchSnapshot("Mounted with default theme.");
  });
});
