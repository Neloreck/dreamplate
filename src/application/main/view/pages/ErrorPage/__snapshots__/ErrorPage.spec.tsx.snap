// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error page component rendering. Should match HOC snapshot.: Default HOC layout. 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Error page component rendering. Should match mounted snapshot with default theme.: Mounted with default theme. 1`] = `
<ErrorPage
  classes={
    Object {
      "content": "ErrorPage-content-0-2-1",
      "label": "ErrorPage-label-0-2-2",
    }
  }
>
  <main
    className="ErrorPage-content-0-2-1"
  >
    <div
      className="ErrorPage-label-0-2-2"
    >
      Page not found.
    </div>
    <button
      onClick={[Function]}
    >
      Home
    </button>
  </main>
</ErrorPage>
`;
